src/index.ts

```diff
@@ -137,14 +137,11 @@ export class LinkChecker extends EventEmitter {
       return;
     }
 
-    // Check for user configured links that should be skipped
-    const skips = opts.checkOptions
-      .linksToSkip!.map(linkToSkip => {
-        return new RegExp(linkToSkip).test(opts.url.href);
-      })
-      .filter(match => !!match);
-
-    if (skips.length > 0) {
+    // Check for a user-configured function to filter out links
+    if (
+      typeof opts.checkOptions.linksToSkip === 'function' &&
```

Comment:
So this is checking if the array itself is a function.  What we want to do is iterate over the items *in* the array, and do a type check on each to determine if that element is a string, regex, or function.  Based on the type of the element, we can react appropriately.

Reply:
I'm not sure I follow you here. I was thinking that `linksToSkip` should be a function OR an array.

Are you suggesting that `linksToSkip` should always be an array, and that its values can be strings, regexes, or functions?

Reply:
Exactly!

Reply:
😅Sounds good.